common/src/main/java/org/conscrypt/ClientSessionContext.java

```diff
@@ -26,39 +28,46 @@
  * @hide
  */
 @Internal
-public class ClientSessionContext extends AbstractSessionContext {
-
+public final class ClientSessionContext extends AbstractSessionContext {
     /**
      * Sessions indexed by host and port. Protect from concurrent
      * access by holding a lock on sessionsByHostAndPort.
      */
-    private final HashMap<HostAndPort, SSLSession> sessionsByHostAndPort = new HashMap<>();
+    @SuppressWarnings("serial")
+    private final HashMap<HostAndPort, SslSessionWrapper> sessionsByHostAndPort =
```

Comment:
Either `Map` or `LinkedHashMap` would be a better choice for the field type here.